diff --git a/node_modules/typescript/lib/typescript.d.ts b/node_modules/typescript/lib/typescript.d.ts
index fbb3e49..7de1e6a 100644
--- a/node_modules/typescript/lib/typescript.d.ts
+++ b/node_modules/typescript/lib/typescript.d.ts
@@ -6088,6 +6088,7 @@ declare namespace ts {
         getIndexInfoOfType(type: Type, kind: IndexKind): IndexInfo | undefined;
         getIndexInfosOfType(type: Type): readonly IndexInfo[];
         getIndexInfosOfIndexSymbol: (indexSymbol: Symbol) => IndexInfo[];
+        getTypeArgumentsForResolvedSignature(signature: Signature): readonly Type[] | undefined;
         getSignaturesOfType(type: Type, kind: SignatureKind): readonly Signature[];
         getIndexTypeOfType(type: Type, kind: IndexKind): Type | undefined;
         getBaseTypes(type: InterfaceType): BaseType[];
diff --git a/node_modules/typescript/lib/typescript.js b/node_modules/typescript/lib/typescript.js
index 714953f..17c90b5 100644
--- a/node_modules/typescript/lib/typescript.js
+++ b/node_modules/typescript/lib/typescript.js
@@ -49956,6 +49956,7 @@ function createTypeChecker(host) {
     getGlobalDiagnostics,
     getRecursionIdentity,
     getUnmatchedProperties,
+    getTypeArgumentsForResolvedSignature,
     getTypeOfSymbolAtLocation: (symbol, locationIn) => {
       const location = getParseTreeNode(locationIn);
       return location ? getTypeOfSymbolAtLocation(symbol, location) : errorType;
@@ -92217,6 +92218,9 @@ function createTypeChecker(host) {
     Debug.assert(specifier && nodeIsSynthesized(specifier) && specifier.text === "tslib", `Expected sourceFile.imports[0] to be the synthesized tslib import`);
     return specifier;
   }
+  function getTypeArgumentsForResolvedSignature(signature) {
+    return signature.mapper && instantiateTypes((signature.target ?? signature).typeParameters ?? [], signature.mapper);
+  }
 }
 function isNotAccessor(declaration) {
   return !isAccessor(declaration);
